packages/core/src/model.types.ts

```diff
@@ -28,20 +30,25 @@ export interface Model<
     eventType?: TEventType
   ) => AssignAction<TContext, ExtractEvent<TEvent, TEventType>>;
   events: Prop<TModelCreators, 'events'>;
+  actions: Prop<TModelCreators, 'actions'>;
   reset: () => AssignAction<TContext, any>;
   createMachine: (
-    config: MachineConfig<TContext, any, TEvent>,
-    implementations?: Partial<MachineOptions<TContext, TEvent>>
-  ) => StateMachine<TContext, any, TEvent>;
+    config: MachineConfig<TContext, any, TEvent, TActions>,
+    implementations?: Partial<MachineOptions<TContext, TEvent, TActions>>
+  ) => StateMachine<TContext, any, TEvent, any>;
```

Comment:
This has brought back the problem that got fixed here: https://github.com/statelyai/xstate/pull/2436

Reply:
I've pushed out a fix (and a test) for this: https://github.com/statelyai/xstate/pull/2266/commits/2f2895b03ac0871b6053cd4ffc5f6d2c872f98e4